### PR TITLE
refactor: change behaviour of `noPeers` event

### LINKED ISSUE #1622: What is the noPeers event expected behavior?
**What version of WebTorrent?**
@latest
**What operating system and Node.js version?**
@latest 

Reproductible snippet : 
```JAVASCRIPT
const wt = require('webtorrent');
var sintel = 'magnet:?xt=urn:btih:08ada5a7a6183aae1e09d831df6748d566095a10&dn=Sintel&tr=udp%3A%2F%2Fexplodie.org%3A6969&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Ftracker.empire-js.us%3A1337&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337&tr=wss%3A%2F%2Ftracker.btorrent.xyz&tr=wss%3A%2F%2Ftracker.fastcast.nz&tr=wss%3A%2F%2Ftracker.openwebtorrent.com&ws=https%3A%2F%2Fwebtorrent.io%2Ftorrents%2F';
var client = new wt();
var tor = client.add(sintel);
tor.on('noPeers', a => console.log('No '+a+' peers'));
```

Since the 'Sintel' torrent is a very healthy torrent with many active trackers and peers sharing it, I don't understand what would be the use of 'noPeers' if it's called on every announce with no peers. 

Shouldn't the event only be called on edge cases when there are no active peers to continue download of the file, or on a very 'unhealthy' torrent with no peers at all to fetch metadata? This behavior makes the event almost useless in every case, unless I'm understanding it wrong. Thanks!

### PR DESCRIPTION
**Pull request by alxhotel**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[X] New feature
[ ] Other, please explain:

**What changes did you make? (Give an overview)**
This PR:
- Changes the behaviour of the `noPeers` event
- Adds a new option to the client constructor to disable `ut_pex` discovery mechanism
- Refactor `Peer` constants

**Which issue (if any) does this pull request address?**
Closes #1622 #2110

**Is there anything you'd like reviewers to focus on?**

**Note**: This is a breaking change so it should be merged to v2 branch.

### CODE REVIEW

File: lib/torrent.js [hunk 1]
```diff
@@ -48,6 +48,8 @@ const PIPELINE_MAX_DURATION = 1
 const RECHOKE_INTERVAL = 10000 // 10 seconds
 const RECHOKE_OPTIMISTIC_DURATION = 2 // 30 seconds
 
+const NO_PEERS_INTERVAL = 30_000 // 30 seconds
```
Comment (DiegoRBaquero) [inline lib/torrent.js:51]: shouldn't this be an opt in the client? defaulted to 30 seconds

**Review by DiegoRBaquero — APPROVED** [review]
I don't think this needs to be a breaking change major version, at least one event is being emitted now

File: lib/torrent.js [hunk 2]
```diff
@@ -94,6 +96,8 @@ class Torrent extends EventEmitter {
     this._rechokeOptimisticWire = null
     this._rechokeOptimisticTime = 0
     this._rechokeIntervalId = null
+    this._noPeersIntervalId = null
+    this._noPeersIntervalTime = opts.noPeersIntervalTime || DEFAULT_NO_PEERS_INTERVAL
```
Comment (DiegoRBaquero) [inline lib/torrent.js:100]:
```suggestion
    this._noPeersIntervalTime = opts.noPeersIntervalTime ? opts.noPeersIntervalTime * 1000 : DEFAULT_NO_PEERS_INTERVAL
```

**Review by DiegoRBaquero — CHANGES_REQUESTED** [review]
opts states seconds, so we must multiply

File: lib/torrent.js (hunk 2 above)
Comment (alxhotel) [reply lib/torrent.js:100]: Oh good catch!

**Review by DiegoRBaquero — APPROVED** [review]

**Comment by ThaUnknown** [comment]
why do incoming peers not have a source?

**Pull request merged**